src/overlay_12_0224E4FC.c

```diff
@@ -0,0 +1,1224 @@
+            } else {
+                temp += data;
+            }
+            
+            mon->friendship = temp;
+        }
+        break;
+    case BMON_DATA_HP:
+        {
+            if (mon->hp + data > mon->maxHp) {
+                mon->hp = mon->maxHp;
+            } else {
+                mon->hp += data;
+            }
+        }
+        break;
+    case BMON_DATA_MAXHP:
+        mon->maxHp += data;
+        break;
+    case BMON_DATA_EXP:
+        mon->exp += data;
+        break;
+    case BMON_DATA_PERSONALITY:
+        mon->personality += data;
+        break;
+    case BMON_DATA_DISABLED_TURNS:
+        mon->unk88.disabledTurns += data;
+        break;
+    case BMON_DATA_ENCORED_TURNS:
+        mon->unk88.encoredTurns += data;
+        break;
+    case BMON_DATA_IS_CHARGED:
+        mon->unk88.isCharged += data;
+        break;
+    case BMON_DATA_TAUNT_TURNS:
+        mon->unk88.tauntTurns += data;
+        break;
+    case BMON_DATA_PROTECT_SUCCESS_COUNT:
+        mon->unk88.protectSuccessTurns += data;
+        break;
+    case BMON_DATA_PERISH_SONG_TURNS:
+        mon->unk88.perishSongTurns += data;
+        break;
+    case BMON_DATA_ROLLOUT_TURNS:
+        mon->unk88.rolloutCount += data;
+        break;
+    case BMON_DATA_FURY_CUTTER_TURNS:
+        mon->unk88.furyCutterCount += data;
+        break;
+    case BMON_DATA_STOCKPILE_COUNT:
+        mon->unk88.stockpileCount += data;
+        break;
+    case BMON_DATA_STOCKPILE_DEF_BOOSTS:
+        mon->unk88.stockpileDefCount += data;
+        break;
+    case BMON_DATA_STOCKPILE_SPDEF_BOOSTS:
+        mon->unk88.stockpileSpDefCount += data;
+        break;
+    case BMON_DATA_LAST_RESORT_COUNT:
+        mon->unk88.lastResortCount += data;
+        break;
+    case BMON_DATA_MAGNET_RISE:
+        mon->unk88.magnetRiseTurns += data;
+        break;
+    case BMON_DATA_HEAL_BLOCK:
+        mon->unk88.healBlockTurns += data;
+        break;
+    case BMON_DATA_RECHARGE:
+        mon->unk88.rechargeCount += data;
+        break;
+    case BMON_DATA_FAKE_OUT:
+        mon->unk88.fakeOutCount += data;
+        break;
+    case BMON_DATA_SLOW_START_COUNT:
+        mon->unk88.slowStartTurns += data;
+        break;
+    case BMON_DATA_SUBSTITUTE_HP:
+        mon->unk88.substituteHp += data;
+        break;
+    case BMON_DATA_HELD_ITEM_RESTORE_HP:
+        mon->unk88.unk30 += data;
+        break;
+    case BMON_DATA_SLOW_START_FLAG:
+        mon->slowStartFlag += data;
+        break;
+    case BMON_DATA_SLOW_START_END:
+        mon->slowStartEnded += data;
+        break;
+    case BMON_DATA_FORME:
+        mon->forme += data;
+        break;
+    default:
+        GF_ASSERT(FALSE);
+    }
+}
+
+extern u8 sStatChangeTable[13][2];
+extern u8 sSpeedHalvingItemEffects[8];
+
+u8 ov12_0224FC48(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId1, int battlerId2, int flag) {
+    u8 ret = 0; //0 - don't sort, 1 - sort, 2 - sort (speed tie + won random check)
```

Comment:
enum/constants?

Reply:
Will once I get to the code that calls this (it's for sorting mons based on speed)